dbgpt/storage/knowledge_graph/community_summary.py

```diff
@@ -138,145 +141,90 @@ def get_config(self) -> BuiltinKnowledgeGraphConfig:
-                        },
-                        edge_name=GraphElemType.INCLUDE.value,
-                        edge_type=GraphElemType.CHUNK_INCLUDE_CHUNK.value,
-                    )
-                else:
-                    # document -> include -> chunk
-                    graph_of_all.upsert_vertex_and_edge(
-                        src_vid=doc_vid,
-                        src_name=doc_name,
-                        src_props={
-                            "vertex_type": GraphElemType.DOCUMENT.value,
-                            "content": "",
-                        },
-                        dst_vid=chunk["id"],
-                        dst_name=chunk["title"],
-                        dst_props={
-                            "vertex_type": GraphElemType.CHUNK.value,
-                            "content": chunk["content"],
-                        },
-                        edge_name=GraphElemType.INCLUDE.value,
-                        edge_type=GraphElemType.DOCUMENT_INCLUDE_CHUNK.value,
+            for chunk_index, chunk in enumerate(chunks):
+                if chunk.chunk_parent_id != "document":
+                    graph_of_all.upsert_document_include_chunk(
+                        chunk=chunk, doc_vid=doc_vid
                     )
 
                 # chunk -> next -> chunk
                 if chunk_index >= 1:
-                    graph_of_all.upsert_vertex_and_edge(
-                        src_vid=data_list[chunk_index - 1]["id"],
-                        src_name=data_list[chunk_index - 1]["title"],
-                        src_props={
-                            "vertex_type": GraphElemType.CHUNK.value,
-                            "content": data_list[chunk_index - 1]["content"],
-                        },
-                        dst_vid=chunk["id"],
-                        dst_name=chunk["title"],
-                        dst_props={
-                            "vertex_type": GraphElemType.CHUNK.value,
-                            "content": chunk["content"],
-                        },
-                        edge_name=GraphElemType.NEXT.value,
-                        edge_type=GraphElemType.CHUNK_NEXT_CHUNK.value,
+                    graph_of_all.upsert_chunk_next_chunk(
+                        chunk=chunks[chunk_index - 1], next_chunk=chunk
                     )
 
         # Support knowledge graph search by the entities and the relationships
-        if self._graph_store.get_config().enable_triplet_graph:
-            for chunk_index, chunk in enumerate(data_list):
+        if triplet_graph_enabled:
+            for chunk_index, chunk in enumerate(chunks):
                 # TODO: Use asyncio to extract graph to accelerate the process
                 # (attention to the CAP of the graph db)
 
                 graphs: List[MemoryGraph] = await self._graph_extractor.extract(
-                    chunk["content"]
+                    chunk.content
```

Comment:
`extract` is the core logic of triple KG. Why is it placed in `_aload_document_graph`?

Reply:
DONE